### PR TITLE
Jacob/rr 103 automated request test to test the email verification flow

### PR DESCRIPTION
**Pull request by jacobperia**
## 📜 Description
- Added automated tests for registration and confirmation flow
- Redirect away from `confirmation page` and also hide `request confirmation email link` if confirmation is disabled

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -143,7 +143,7 @@
   # without confirming their account.
   # Default is 0.days, meaning the user cannot access the website without
   # confirming their account.
-  config.allow_unconfirmed_access_for = 0.days
```
Comment (jacobperia) [inline config/initializers/devise.rb:146]: this can be commented out because it's a default value anyways

**Comment by danielpaul** [comment]
@jacobperia the CI here (GitHub) seems to failing

**Comment by jacobperia** [comment]
> @jacobperia the CI here (GitHub) seems to failing

@danielpaul looks like some css issue? It's passing on my local though.

**Comment by jacobperia** [comment]
> @jacobperia the CI here (GitHub) seems to failing

@danielpaul Looks like some CSS issue with the tailwind include tag not related to my changes? Also, all tests are passing on my local.

File: config/initializers/devise.rb (hunk 1 above)
Comment (danielpaul) [reply config/initializers/devise.rb:146]: let's have this commented in so it's clear that this is something the user can customise. in the docs we just say that they need to update this so we don't want to make the docs more complex where we need to say to comment it in and then customise or have cause for confusion around this.

**Review by danielpaul — CHANGES_REQUESTED** [review]

**Comment by danielpaul** [comment]
hey @jacobperia rebasing this with main so we can see if the CI will pass. CI should be passing in `main`

**Comment by jacobperia** [comment]
> hey @jacobperia rebasing this with main so we can see if the CI will pass. CI should be passing in `main`

the css issue still seems to be present?

File: spec/requests/registrations_spec.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  def post_request(path, params)
+    post path, params:
+  end
+
+  describe "Email verification flow" do
+    before :each do
+      Devise.allow_unconfirmed_access_for = 0.days
+    end
+
+    context "when xonfirmation is required setting is enabled in devise config" do
```
Comment (danielpaul) [inline spec/requests/registrations_spec.rb:12]:
```suggestion
    context "when confirmation is required setting is enabled in devise config" do
```

**Comment by danielpaul** [comment]
@jacobperia I have rebased this from `main`. The `main` branch is passing `rspec` while this isn't. It must be a change that was introduced in this branch? But it is a strange error that can't seem to see how the CSS one would be related to the changes in this PR. 🤔

**Comment by jacobperia** [comment]
> @jacobperia I have rebased this from `main`. The `main` branch is passing `rspec` while this isn't. It must be a change that was introduced in this branch? But it is a strange error that can't seem to see how the CSS one would be related to the changes in this PR. 🤔

thats what's confusing me too 😕

**Comment by jacobperia** [comment]
> > @jacobperia I have rebased this from `main`. The `main` branch is passing `rspec` while this isn't. It must be a change that was introduced in this branch? But it is a strange error that can't seem to see how the CSS one would be related to the changes in this PR. 🤔
> 
> It seems that this is related to this test. Commenting this out seems to make it work.
> 
> CleanShot 2023-08-17 at 16 08 38@2x
> Is there anything else here that we are changing for this test that is messing with CSS?

nope. i think since it's a syntax error, it's probably something that would show up on the rails console if we visit that page normally also. It's just not going to throw a hard error.

**Comment by jacobperia** [comment]
> > @jacobperia I have rebased this from `main`. The `main` branch is passing `rspec` while this isn't. It must be a change that was introduced in this branch? But it is a strange error that can't seem to see how the CSS one would be related to the changes in this PR. 🤔
> 
> It seems that this is related to this test. Commenting this out seems to make it work.
> 
> CleanShot 2023-08-17 at 16 08 38@2x
> Is there anything else here that we are changing for this test that is messing with CSS?

@danielpaul not really...also, it's not easy to debug this since it's only happening on CI. On my local there are no issues and all tests pass.

**Pull request merged**